### PR TITLE
Neo4j integration

### PR DESCRIPTION
**Pull request by auggie246**
This PR brings in changes to the terraform scripts to include an additional neo4j module that can be toggled by the user to set up a neo4j instance. In addition, the required helm charts were added to the `helm-charts` folder

1. A new argument in `.tfvars` to toggle neo4j module deployment
2. 2 helm charts from official neo4j repo has been added. Most of the repo is unchanged except from `values.yaml` where some values were added to set as default for AISG's deployment. 
3. The `neo4j` chart is to deploy the base neo4j instance while `neo4j-reverse-proxy` creates an HTTP server, which routes requests to either the Bolt reverse proxy or HTTP reverse proxy based on the request headers. This is required to deploy Neo4j with ingress on GKE.
4. `values.yaml` and `reverse-proxy-values.yaml` from` `terraform/modules/neo4j` contain some values that should be modified by the user to override certain key fields from the Helm repo!

### CODE REVIEW

**Comment by auggie246** [comment]
Once vetted, please hold before merging, I have to remove the temp branch reference in [764f4cf](https://github.com/aisingapore/kapitan-hull-admin/pull/15/commits/764f4cf66729d661b5153be490ff22bd1fddfe11)

**Comment by auggie246** [comment]
@deonchia @andyozj @Syakyr for your review

File: terraform/envs/gcp/main.tf [hunk 1]
```diff
@@ -81,3 +81,13 @@ module "coder-server" {
   oidc_client_id       = var.coder_auth == "oidc" ? var.oidc_client_id : null
   oidc_client_secret   = var.coder_auth == "oidc" ? var.oidc_client_secret : null
 }
+
+module "neo4j" {
+  source               = "github.com/aisingapore/kapitan-hull-admin//terraform/modules/neo4j?ref=neo4j-integration"
```
Comment (deonchia) [inline terraform/envs/gcp/main.tf]: If we're pulling directly from github, best to set to a shallow clone, i.e. `depth=1`

File: terraform/envs/gcp/gcp.tfvars [hunk 2]
```diff
@@ -13,3 +13,4 @@ oidc_email_domain  = "aisingapore.org\\,aiap.sg"
 oidc_client_id     = "OIDC_CLIENT_ID"
 oidc_client_secret = "OIDC_CLIENT_SECRET"
 
+enable_neo4j = false
```
Comment (deonchia) [inline terraform/envs/gcp/gcp.tfvars]: I think `neo4j = true|false` would suffice

**Review by deonchia — CHANGES_REQUESTED** [review]
Good stuff! Generally okay, but will definitely need to include READMEs for possible values for both `values.yaml` and `variables.tf`

File: terraform/envs/gcp/main.tf (hunk 1 above)
Comment (auggie246) [reply terraform/envs/gcp/main.tf]: Added in [dac817a](https://github.com/aisingapore/kapitan-hull-admin/pull/15/commits/dac817a214dacb7290287c642adb85fccbcb6b9d)

File: terraform/envs/gcp/gcp.tfvars (hunk 2 above)
Comment (auggie246) [reply terraform/envs/gcp/gcp.tfvars]: Modified in [b36784c](https://github.com/aisingapore/kapitan-hull-admin/pull/15/commits/b36784c48f00b5795b10f1a87182c2ef9a31fe78)

**Comment by auggie246** [comment]
> Good stuff! Generally okay, but will definitely need to include READMEs for possible values for both `values.yaml` and `variables.tf`

Created a readme at [43bc731](https://github.com/aisingapore/kapitan-hull-admin/pull/15/commits/43bc731c6543cf49c6eab09b7561f98fe20064ad)

**Review by deonchia — APPROVED** [review]
LGTM!

**Pull request merged**